autoload/phpactor/project/project.vim

```diff
@@ -0,0 +1,39 @@
+function! phpactor#project#project#createFromRootPath(primaryRootPath) abort
+  let l:project = {
+        \ 'primaryRootPath': s:normalizeRootPath(a:primaryRootPath),
+        \ 'getPrimaryRootPath': function('s:getPrimaryRootPath'),
+        \ 'containsFile': function('s:containsFile')
+        \ }
+
+  return l:project
+endfunction
+
+function! s:getPrimaryRootPath() dict
+  return l:self.primaryRootPath
+endfunction
+
+function! s:containsFile(filename) dict
```

Comment:
`finddir()` and `filepath()` might be used here instead if we have full path and not just a filename as parameter.
If so maybe rename it at the same time to avoid confusion ;)

Reply:
I outsourced most of directory tree logic into `phpactor#fileutils` and tried to use those functions where it could be possible.